superset-frontend/src/features/databases/DatabaseModal/DatabaseConnectionForm/index.tsx

```diff
@@ -161,7 +65,7 @@ const DatabaseConnectionForm = ({
               Object.keys(parameters.properties).includes(key) ||
               key === 'database_name',
           ).map(field =>
-            FORM_FIELD_MAP[field]({
+            (FORM_FIELD_MAP as any)[field]({
```

Comment:
Do we need that `as any`? If typescript is complaining, maybe we should rather cast field as one of `FORM_FIELD_MAP`'s keys?

Reply:
This was just a quick-fix from copilot for a `ts` error that is negligible in the scope of this PR and a house of cards for typing incompatibility throughout the database modal so i'm going to revert it for now and maybe the type definitions can be revised in a later PR.